pkg/storage/store.go

```diff
@@ -2442,6 +2442,7 @@ func (s *Store) Send(
 			if pErr != nil {
 				pErr.OriginNode = ba.Replica.NodeID
 				if txn := pErr.GetTxn(); txn != nil {
+					// Clone the txn, as we'll modify it.
 					pErr.SetTxn(txn)
```

Comment:
Tamir, is cloning inside pErr really necessary (as opposed to cloning ba.Txn)? Is the error shared?